src/main/groovy/com/twcable/grabbit/jcr/JCRNodeDecorator.groovy

```diff
@@ -105,6 +110,57 @@ class JCRNodeDecorator {
         return definition.isMandatory()
     }
 
+    /**
+     * Build node and "only" mandatory child nodes
+     * @param decoratedNode
+     * @param mandatoryChildNodes
+     */
+    @Nonnull
+     ProtoNode buildNode() {
+        final ProtoNodeBuilder protoNodeBuilder = ProtoNode.newBuilder()
+        protoNodeBuilder.setName(path)
+        protoNodeBuilder.addAllProperties(getProtoProperties())
+        requiredChildNodes?.each {
+            protoNodeBuilder.addMandatoryChildNode(it.buildNode())
+        }
+        return protoNodeBuilder.build()
+    }
+
+    /**
+     * Build a collection of Proto Properties from JCR node
+     * @param decoratedNode to gather properties from
+     * @return resulting collection of transferable proto properties collected from jcrNode
+     */
+    @Nonnull
+    Collection<ProtoProperty> getProtoProperties() {
+        final List<Property> properties = properties.toList()
+        return properties.findResults { JcrProperty jcrProperty ->
+            JcrPropertyDecorator decoratedProperty = new JcrPropertyDecorator(jcrProperty);
+            decoratedProperty.isTransferable() ? decoratedProperty.toProtoProperty() : null ;
+        }
+    }
+
+    /**
+     * Returns the "jcr:created", "jcr:lastModified" or "cq:lastModified" date property
+     * for current Jcr Node
+     * @param jcrNode
+     * @return null if none of the 3 are found
+     */
+    @Nullable
+    Date getDate() {
```

Comment:
Sure...I figure if we are describing what the method does, the last modified date can be the created date..